### PR TITLE
Shim python interpreter for windows if interpreter uses reparse points

### LINKED ISSUE #5388: Windows 10: Failed to locate any of ["python3" "python"] on your PATH
When I try to run `pulumi new aws-python` on Windows 10, I get the error `Failed to locate any of ["python3" "python"] on your PATH.  Have you installed Python 3.6 or greater?` 

I can type `python --version` into the command prompt which returns `Python 3.7.9` so this do the trick...

## Expected Behavior
It should create the project scaffolding

## Current Behavior
Displays the `Failed to locate any of ["python3" "python"] on your PATH.  Have you installed Python 3.6 or greater` error message

## Steps to Reproduce

1. python --version
2. mkdir quickstart
3. cd quickstart
4. pulumi new aws-python

## Context (Environment)
Windows 10 with Python installed (from the Microsoft Store) to C:\Users\USERNAME\AppData\Local\Microsoft\WindowsApps

### PR DESCRIPTION
**Pull request by viveklak**
Alternative implementation to #5775 
Fixes #5388 

Uses a shim script to workaround https://github.com/golang/go/issues/42919

### CODE REVIEW

**Review by justinvp — APPROVED** [review]
LGTM otherwise.

One question I have is around our `IsVirtualEnv` and `VirtualEnvCommand` functions.

- `IsVirtualEnv` checks if there's a `python.exe` binary inside the specified virtual environment directory, via call to `os.Stat` looking for `Scripts/python.exe`
- `VirtualEnvCommand` returns an `exec.Cmd` for the binary in `Scripts/<bin>.exe`. This function is used to run `Scripts/python.exe` inside the virtual environment.

Do these work on systems where Python is installed from the app store? (Basically, when we run `pulumi-python-shim.cmd <resolved_execution_alias_python_binary> -m venv venv` to create the virtual environment, in the resulting `venv` directory, what type of file is `venv/Scripts/python.exe`? Can we successfully `os.Stat` and `exec.Cmd` it, or is it one of these execution aliases that needs `Lstat` and shimming?)

A way to test it: With the changes in this PR, create a standard Python Pulumi project via `pulumi new python` and then run `pulumi up`. That should exercise these code paths.

**Comment by viveklak** [comment]
> LGTM otherwise.
> 
> One question I have is around our `IsVirtualEnv` and `VirtualEnvCommand` functions.
> 
> * `IsVirtualEnv` checks if there's a `python.exe` binary inside the specified virtual environment directory, via call to `os.Stat` looking for `Scripts/python.exe`
> * `VirtualEnvCommand` returns an `exec.Cmd` for the binary in `Scripts/<bin>.exe`. This function is used to run `Scripts/python.exe` inside the virtual environment.
> 
> Do these work on systems where Python is installed from the app store? (Basically, when we run `pulumi-python-shim.cmd <resolved_execution_alias_python_binary> -m venv venv` to create the virtual environment, in the resulting `venv` directory, what type of file is `venv/Scripts/python.exe`? Can we successfully `os.Stat` and `exec.Cmd` it, or is it one of these execution aliases that needs `Lstat` and shimming?)
> 
Sorry - I got lazy updating the PR body last night. I did test with the venv setup - i.e. create a new project with python and make sure the python within the venv isn't affected. I was able to do a pulumi up just fine. What I have seen is that the venv's python interpreter is a regular file with a non-zero length so its not affected by the bug.

File: scripts/make_release.ps1 [hunk 1]
```diff
@@ -36,6 +36,7 @@ RunGoBuild "github.com/pulumi/pulumi/sdk/v2/go/pulumi-language-go" "sdk" "pulumi
 CopyPackage "$Root\sdk\nodejs\bin" "pulumi"
 
 Copy-Item "$Root\sdk\nodejs\dist\pulumi-resource-pulumi-nodejs.cmd" "$PublishDir\bin"
+Copy-Item "$Root\sdk\nodejs\dist\pulumi-python-shim.cmd" "$PublishDir\bin"
```
Comment (viveklak) [inline scripts/make_release.ps1:39]: Note that we need the pulumi-python-shim.cmd packaged as part of the plugin and be installed in PATH. Based on what I saw, this should be sufficient but I am not familiar enough with the mechanics of the build/publish for windows to know for sure this works. My crappy windows VM can't handle the build for pulumi so I am leaving this to CI to build and will validate the published zip to check if the script makes it in. FYI @justinvp

File: sdk/python/dist/pulumi-python-shim.cmd [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+set params=%1
+:loop
+shift
+if [%1]==[] goto afterloop
+set params=%params% %1
+goto loop
+:afterloop
+
+%python_interpreter% %params%
```
[the remark this answers is not in the document]
Comment (viveklak) [reply sdk/python/dist/pulumi-python-shim.cmd]: I followed your advice mostly but left the initial command hard coded.

**Comment by viveklak** [comment]
Merging so we can test with dev build.

**Pull request merged**